.buildkite/scripts/steps/build_and_test.sh

```diff
@@ -100,15 +100,15 @@ else
   fi
 fi
 
-if ! [[ "$HARDWARE_ARCH" = aarch64 && -n "$CPP_CROSS_COMPILE" ]] && [[ $TEST_OUTCOME -eq 0 ]] ; then 
-  buildkite-agent artifact upload "build/distributions/*.zip"
-fi
-
 if [[ -z "$CPP_CROSS_COMPILE" ]] ; then 
-  OS=$(uname -s | tr "A-Z" "a-z")
-  TEST_RESULTS_ARCHIVE=${OS}-${HARDWARE_ARCH}-unit_test_results.tgz
-  find . -path  "*/**/ml_test_*.out" -o -path "*/**/*.junit" | xargs tar cvzf ${TEST_RESULTS_ARCHIVE}
-  buildkite-agent artifact upload "${TEST_RESULTS_ARCHIVE}"
+  if [[ $TEST_OUTCOME -eq 0 ]] ; then 
+    buildkite-agent artifact upload "build/distributions/*.zip"
+  else
```

Comment:
I think it's best to upload the test output even if the build succeeds. If we only upload test results when there are failures then we'll never be able to compare successful output against failure output.